File: frontend/src/services/announcementService.js
import api from './api'

/**
 * Fetch all active announcements.
 * @param {string|null} type - Optional filter: 'URGENT' | 'COLLEGE' | 'DEPARTMENT' | 'GENERAL'
 */
export async function getAnnouncements(type = null) {
  const params = type ? { type } : {}
  const response = await api.get('/announcements', { params })
  return response.data.data   // unwrap ApiResponse<List<AnnouncementResponse>>
}

/**
 * Fetch a single announcement by ID.
 */
export async function getAnnouncementById(id) {
  const response = await api.get(`/announcements/${id}`)
  return response.data.data
}

/**
 * Fetch announcement stats (e.g. urgentCount for badge).
 */
export async function getAnnouncementStats() {
  const response = await api.get('/announcements/stats')
  return response.data.data
}

File: backend/src/main/java/com/campusone/service/AnnouncementService.java
package com.campusone.service;

import com.campusone.dto.response.AnnouncementResponse;
import com.campusone.entity.AnnouncementType;

import java.util.List;
import java.util.Optional;

public interface AnnouncementService {

    /** Fetch all active announcements, optionally filtered by type. */
    List<AnnouncementResponse> getAnnouncements(AnnouncementType type);

    /** Fetch a single announcement by ID. */
    Optional<AnnouncementResponse> getById(Long id);

    /** Count currently active urgent announcements. */
    long countActiveUrgent();
}
